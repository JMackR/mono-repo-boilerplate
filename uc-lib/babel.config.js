module.exports = {
  presets: [["@babel/preset-env", { targets: { node: "current" } }], "module:metro-react-native-babel-preset"],
  plugins: ["@babel/plugin-proposal-class-properties", "inline-json-import"],
  sourceType: "unambiguous",
}
