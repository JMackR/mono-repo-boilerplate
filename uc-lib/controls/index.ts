export * from "./image"
export * from "./button"
export * from "./text"
export * from "./activity-indicator"
export * from "./separator"
export * from "./list"
export * from "./section-list"
export * from "./badge"
export * from "./toggle"
export * from "./login-button"
export * from "./layout"
export * from "./background"
export * from "./grid"
export * from "./touchable"
export * from "./view-block"
export * from "./text-entry"
export * from "./scroll-view"
export * from "./circle"
export * from "./slider"
export * from "./clickable"
export * from "./map"
export * from "./visibility-check"
export * from "./strike-through-text"
// export * from "./stripe-text-entry"
export * from "./input"
