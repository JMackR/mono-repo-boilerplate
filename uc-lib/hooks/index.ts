export * from "./use-infinite-scroll"
export * from "./selectable"
export * from "./screen-provider"
export * from "./loading"
