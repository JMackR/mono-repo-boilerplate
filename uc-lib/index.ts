export * from "./animations"
export * from "./assets"
export * from "./controls"
export * from "./themes"
export * from "./widgets"
export * from "./utilities"
export * from "./hooks"
export * from "./validators"
