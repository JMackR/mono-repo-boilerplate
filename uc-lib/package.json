{
  "name": "redibs-ucl",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "test": "jest --projects test/react test/react-native test/non-react"
  },
  "dependencies": {
    "@types/ramda": "^0.27.14",
    "classnames": "2.2.6",
    "google-map-react": "1.1.7",
    "hoist-non-react-statics": "3.3.2",
    "imask": "6.0.5",
    "react-compound-slider": "2.5.0",
    "react-native-autolink": "3.0.0",
    "react-native-fs": "^2.16.6",
    "zxcvbn": "4.4.2"
  },
  "devDependencies": {},
  "prettier": {
    "printWidth": 125,
    "semi": false,
    "singleQuote": false,
    "trailingComma": "all"
  },
  "jest-junit": {
    "outputDirectory": "../build",
    "ancestorSeparator": " â€º ",
    "classNameTemplate": "{classname}",
    "suiteName": "redibs-redibs-ucl jest tests",
    "suiteNameTemplate": "{filename}",
    "titleTemplate": "{title}",
    "uniqueOutputName": "true"
  }
}
