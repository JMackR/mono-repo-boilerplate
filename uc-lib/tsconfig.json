{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "baseUrl": ".",
    "outDir": "lib",
    "rootDir": "../",
    "jsx": "react-native",
    "composite": true,
    "declarationMap": true,
    "isolatedModules": true,
    "sourceMap": true,
    "moduleResolution": "Node",
    "esModuleInterop": true,
    "typeRoots": ["../node_modules/@types", "./types"]
  },
  "include": ["./"],
  "exclude": ["node_modules", "babel.config.js", "metro.config.js", "jest.config.js", "types"]
}
