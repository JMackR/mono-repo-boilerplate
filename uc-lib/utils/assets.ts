import { PixelRatio } from "react-native"

const pixelDensityExtension = () => {
  const pixelRatio = PixelRatio.get()
  if (pixelRatio >= 3) {
    return "@3x"
  } else if (pixelRatio >= 2) {
    return "@2x"
  } else {
    return ""
  }
}

/**
 * Constructs and returns a formatted cdn url
 * @param path to asset including extension (e.g. "mojis/assassin-64.png")
 * @return the constructed cdn url
 */
export const cdnUrl = (path: string) => {
  const prefix = "https://wrkshp-cdn.com/com.nextmusic/"
  const url = `${prefix}${path}`
  const index = url.lastIndexOf(".")
  return url.slice(0, index) + pixelDensityExtension() + url.slice(index)
}
