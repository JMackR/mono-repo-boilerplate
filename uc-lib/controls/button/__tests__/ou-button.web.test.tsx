import React from "react"
import { shallow, ShallowWrapper } from "enzyme"
import { Button } from "uc-lib"
import * as renderer from "react-test-renderer"
import { StyleSheetTestUtils } from "aphrodite/no-important"

import { createSerializer } from "jest-aphrodite-react/no-important"
import { LocationPinFill } from "uc-lib/assets"
expect.addSnapshotSerializer(createSerializer(() => StyleSheetTestUtils, { removeVendorPrefixes: true }))

describe("Button Functionality Tests", () => {
  let wrapper: ShallowWrapper,
    click = jest.fn()
  beforeEach(() => {
    wrapper = shallow(<Button title="test" buttonSize="large" buttonType="primary" onClick={click} />)
  })
  test("Click calls passed in onClick function", () => {
    ;(wrapper.props() as any).onClick({ preventDefault: jest.fn() })
    expect(click).toHaveBeenCalledTimes(1)
  })
})

describe("Button Snapshot Tests", () => {
  test("Primary large button renderer.creates properly", () => {
    const tree = renderer.create(<Button title="test" buttonSize="large" buttonType="primary" onClick={jest.fn()} />)

    expect(tree).toMatchSnapshot()
  })
  test("Secondary large button renderer.creates properly", () => {
    const tree = renderer.create(
      <Button title="testingsecondary" buttonSize="large" buttonType="secondary" onClick={jest.fn()} />,
    )
    expect(tree).toMatchSnapshot()
  })
  test("Tertiary large button renderer.creates properly", () => {
    const tree = renderer.create(<Button title="tert" buttonSize="large" buttonType="tertiary" onClick={jest.fn()} />)

    expect(tree).toMatchSnapshot()
  })
  test("Buy Now large button renderer.creates properly", () => {
    const tree = renderer.create(<Button title="buynow" buttonSize="large" buttonType="buynow" onClick={jest.fn()} />)
    expect(tree).toMatchSnapshot()
  })
  test("Flat large button renderer.creates properly", () => {
    const tree = renderer.create(<Button title="flat" buttonSize="large" buttonType="flat" onClick={jest.fn()} />)
    expect(tree).toMatchSnapshot()
  })
  test("Disabled large button renderer.creates properly", () => {
    const tree = renderer.create(<Button title="disabled" buttonSize="large" buttonType="disabled" onClick={jest.fn()} />)

    expect(tree).toMatchSnapshot()
  })
  test("Primary small button renderer.creates properly", () => {
    const tree = renderer.create(<Button title="test" buttonSize="small" buttonType="primary" onClick={jest.fn()} />)

    expect(tree).toMatchSnapshot()
  })
  test("Secondary small button renderer.creates properly", () => {
    const tree = renderer.create(
      <Button title="testingsecondary" buttonSize="small" buttonType="secondary" onClick={jest.fn()} />,
    )
    expect(tree).toMatchSnapshot()
  })
  test("Tertiary small button renderer.creates properly", () => {
    const tree = renderer.create(<Button title="tert" buttonSize="small" buttonType="tertiary" onClick={jest.fn()} />)

    expect(tree).toMatchSnapshot()
  })
  test("Buy Now small button renderer.creates properly", () => {
    const tree = renderer.create(<Button title="buynow" buttonSize="large" buttonType="buynow" onClick={jest.fn()} />)
    expect(tree).toMatchSnapshot()
  })
  test("Flat large small renderer.creates properly", () => {
    const tree = renderer.create(<Button title="flat" buttonSize="small" buttonType="flat" onClick={jest.fn()} />)
    expect(tree).toMatchSnapshot()
  })
  test("Disabled small button renderer.creates properly", () => {
    const tree = renderer.create(<Button title="disabled" buttonSize="small" buttonType="disabled" onClick={jest.fn()} />)

    expect(tree).toMatchSnapshot()
  })
  test("Primary large button with icon as LocalSVGSource renderer.creates properly", () => {
    const tree = renderer.create(
      <Button title="has svg" buttonSize="large" buttonType="primary" icon={LocationPinFill} onClick={jest.fn()} />,
    )

    expect(tree).toMatchSnapshot()
  })
  test("Primary small button with icon as LocalSVGSource renderer.creates properly", () => {
    const tree = renderer.create(
      <Button title="has svg" buttonSize="small" buttonType="primary" icon={LocationPinFill} onClick={jest.fn()} />,
    )

    expect(tree).toMatchSnapshot()
  })
})
