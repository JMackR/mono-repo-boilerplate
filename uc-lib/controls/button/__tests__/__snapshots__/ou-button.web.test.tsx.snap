// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot Tests Buy Now large button renderer.creates properly 1`] = `
.textContainer {
  display: inline-flex;
  flex-direction: column;
  vertical-align: middle;
}

.text {
  color: #121212;
  flex: 1;
  font-family: Lato;
  font-size: 16px;
  font-weight: bold;
  text-align: center;
}

.text:hover {
  color: #121212;
}

.text:active {
  color: #121212;
}

.button {
  padding-left: 10px;
  padding-right: 10px;
  height: 40px;
  background-color: #F4BE19;
  border-radius: 4px;
  border-style: solid;
  border-color: #00000000;
  border-width: 1px;
  box-sizing: border-box;
  text-decoration: none;
  justify-content: center;
  display: flex;
  align-items: center;
  line-height: 24px;
  user-select: none;
}

.button:hover {
  background-color: #F5D624;
  cursor: pointer;
}

.button:active {
  background-color: #F3A70C;
}

<div
  className="button"
  onClick={[Function]}
>
  <div
    className="textContainer"
  >
    <span
      className="text"
    >
      buynow
    </span>
  </div>
</div>
`;

exports[`Button Snapshot Tests Buy Now small button renderer.creates properly 1`] = `
.textContainer {
  display: inline-flex;
  flex-direction: column;
  vertical-align: middle;
}

.text {
  color: #121212;
  flex: 1;
  font-family: Lato;
  font-size: 16px;
  font-weight: bold;
  text-align: center;
}

.text:hover {
  color: #121212;
}

.text:active {
  color: #121212;
}

.button {
  padding-left: 10px;
  padding-right: 10px;
  height: 40px;
  background-color: #F4BE19;
  border-radius: 4px;
  border-style: solid;
  border-color: #00000000;
  border-width: 1px;
  box-sizing: border-box;
  text-decoration: none;
  justify-content: center;
  display: flex;
  align-items: center;
  line-height: 24px;
  user-select: none;
}

.button:hover {
  background-color: #F5D624;
  cursor: pointer;
}

.button:active {
  background-color: #F3A70C;
}

<div
  className="button"
  onClick={[Function]}
>
  <div
    className="textContainer"
  >
    <span
      className="text"
    >
      buynow
    </span>
  </div>
</div>
`;

exports[`Button Snapshot Tests Disabled large button renderer.creates properly 1`] = `
.textContainer {
  display: inline-flex;
  flex-direction: column;
  vertical-align: middle;
}

.button {
  padding-left: 10px;
  padding-right: 10px;
  height: 40px;
  background-color: #FAFAFA;
  border-radius: 4px;
  border-style: solid;
  border-color: #E2E2E2;
  border-width: 1px;
  box-sizing: border-box;
  text-decoration: none;
  justify-content: center;
  display: flex;
  align-items: center;
  line-height: 24px;
  user-select: none;
}

.button:hover {
  background-color: #FAFAFA;
  cursor: pointer;
}

.button:active {
  background-color: #FAFAFA;
}

.text {
  color: #ABABAB;
  flex: 1;
  font-family: Lato;
  font-size: 16px;
  font-weight: bold;
  text-align: center;
}

.text:hover {
  color: #ABABAB;
}

.text:active {
  color: #ABABAB;
}

<div
  className="button"
  onClick={[Function]}
>
  <div
    className="textContainer"
  >
    <span
      className="text"
    >
      disabled
    </span>
  </div>
</div>
`;

exports[`Button Snapshot Tests Disabled small button renderer.creates properly 1`] = `
.textContainer {
  display: inline-flex;
  flex-direction: column;
  vertical-align: middle;
}

.button {
  padding-left: 6px;
  padding-right: 6px;
  height: 32px;
  background-color: #FAFAFA;
  border-radius: 4px;
  border-style: solid;
  border-color: #E2E2E2;
  border-width: 1px;
  box-sizing: border-box;
  text-decoration: none;
  justify-content: center;
  display: flex;
  align-items: center;
  line-height: 24px;
  user-select: none;
}

.button:hover {
  background-color: #FAFAFA;
  cursor: pointer;
}

.button:active {
  background-color: #FAFAFA;
}

.text {
  color: #ABABAB;
  flex: 1;
  font-family: Lato;
  font-size: 14px;
  font-weight: bold;
  text-align: center;
}

.text:hover {
  color: #ABABAB;
}

.text:active {
  color: #ABABAB;
}

<div
  className="button"
  onClick={[Function]}
>
  <div
    className="textContainer"
  >
    <span
      className="text"
    >
      disabled
    </span>
  </div>
</div>
`;

exports[`Button Snapshot Tests Flat large button renderer.creates properly 1`] = `
.textContainer {
  display: inline-flex;
  flex-direction: column;
  vertical-align: middle;
}

.text {
  color: #00A87E;
  flex: 1;
  font-family: Lato;
  font-size: 16px;
  font-weight: bold;
  text-align: center;
}

.text:hover {
  color: #43B893;
}

.text:active {
  color: #00986A;
}

.button {
  padding-left: 10px;
  padding-right: 10px;
  height: 40px;
  background-color: #FFFFFF00;
  border-radius: 4px;
  border-style: solid;
  border-color: #00000000;
  border-width: 1px;
  box-sizing: border-box;
  text-decoration: none;
  justify-content: center;
  display: flex;
  align-items: center;
  line-height: 24px;
  user-select: none;
}

.button:hover {
  background-color: #F7F7F7;
  cursor: pointer;
}

.button:active {
  background-color: #EEEEEE;
}

<div
  className="button"
  onClick={[Function]}
>
  <div
    className="textContainer"
  >
    <span
      className="text"
    >
      flat
    </span>
  </div>
</div>
`;

exports[`Button Snapshot Tests Flat large small renderer.creates properly 1`] = `
.textContainer {
  display: inline-flex;
  flex-direction: column;
  vertical-align: middle;
}

.text {
  color: #00A87E;
  flex: 1;
  font-family: Lato;
  font-size: 14px;
  font-weight: bold;
  text-align: center;
}

.text:hover {
  color: #43B893;
}

.text:active {
  color: #00986A;
}

.button {
  padding-left: 6px;
  padding-right: 6px;
  height: 32px;
  background-color: #FFFFFF00;
  border-radius: 4px;
  border-style: solid;
  border-color: #00000000;
  border-width: 1px;
  box-sizing: border-box;
  text-decoration: none;
  justify-content: center;
  display: flex;
  align-items: center;
  line-height: 24px;
  user-select: none;
}

.button:hover {
  background-color: #F7F7F7;
  cursor: pointer;
}

.button:active {
  background-color: #EEEEEE;
}

<div
  className="button"
  onClick={[Function]}
>
  <div
    className="textContainer"
  >
    <span
      className="text"
    >
      flat
    </span>
  </div>
</div>
`;

exports[`Button Snapshot Tests Primary large button renderer.creates properly 1`] = `
.button {
  padding-left: 10px;
  padding-right: 10px;
  height: 40px;
  background-color: #00A87E;
  border-radius: 4px;
  border-style: solid;
  border-color: #00000000;
  border-width: 1px;
  box-sizing: border-box;
  text-decoration: none;
  justify-content: center;
  display: flex;
  align-items: center;
  line-height: 24px;
  user-select: none;
}

.button:hover {
  background-color: #43B893;
  cursor: pointer;
}

.button:active {
  background-color: #00986A;
}

.textContainer {
  display: inline-flex;
  flex-direction: column;
  vertical-align: middle;
}

.text {
  color: #FFFFFF;
  flex: 1;
  font-family: Lato;
  font-size: 16px;
  font-weight: bold;
  text-align: center;
}

.text:hover {
  color: #F7F7F7;
}

.text:active {
  color: #EEEEEE;
}

<div
  className="button"
  onClick={[Function]}
>
  <div
    className="textContainer"
  >
    <span
      className="text"
    >
      test
    </span>
  </div>
</div>
`;

exports[`Button Snapshot Tests Primary large button with icon as LocalSVGSource renderer.creates properly 1`] = `
.button {
  padding-left: 10px;
  padding-right: 10px;
  height: 40px;
  background-color: #00A87E;
  border-radius: 4px;
  border-style: solid;
  border-color: #00000000;
  border-width: 1px;
  box-sizing: border-box;
  text-decoration: none;
  justify-content: center;
  display: flex;
  align-items: center;
  line-height: 24px;
  user-select: none;
}

.button:hover {
  background-color: #43B893;
  cursor: pointer;
}

.button:active {
  background-color: #00986A;
}

.textContainer {
  display: inline-flex;
  flex-direction: column;
  vertical-align: middle;
}

.text {
  color: #FFFFFF;
  flex: 1;
  font-family: Lato;
  font-size: 16px;
  font-weight: bold;
  text-align: center;
}

.text:hover {
  color: #F7F7F7;
}

.text:active {
  color: #EEEEEE;
}

.iconContainer {
  padding-right: 8px;
  display: inline-flex;
  vertical-align: middle;
}

.icon {
  height: 20px;
  width: 20px;
}

<div
  className="button"
  onClick={[Function]}
>
  <div
    className="iconContainer"
  >
    <span
      className="icon"
    >
      <location-fill-pin.svg
        alwaysdark="#121212"
        alwayslight="#FFFFFF"
        basalt="#8A8A8A"
        crystal="#FFFFFF"
        crystalhover="#F7F7F7"
        crystalpressed="#EEEEEE"
        disabled="#FAFAFA"
        glacialblue="#3D98D2"
        granite="#ABABAB"
        height={20}
        larchyellow="#F4BE19"
        larchyellowhover="#F5D624"
        larchyellowpressed="#F3A70C"
        limestone="#E2E2E2"
        limestonehover="#EEEEEE"
        limestonepressed="#D0D0D0"
        obsidian="#121212"
        overlay="rgba(18,18,18,0.3)"
        paintbrushred="#E05666"
        palousegreen="#00A87E"
        palousegreenhighlight="#DEF3EC"
        palousegreenhover="#43B893"
        palousegreenpressed="#00986A"
        quartz="#FAFAFA"
        tintcolor="#FFFFFF"
        width={20}
      />
    </span>
  </div>
  <div
    className="textContainer"
  >
    <span
      className="text"
    >
      has svg
    </span>
  </div>
</div>
`;

exports[`Button Snapshot Tests Primary small button renderer.creates properly 1`] = `
.textContainer {
  display: inline-flex;
  flex-direction: column;
  vertical-align: middle;
}

.button {
  padding-left: 6px;
  padding-right: 6px;
  height: 32px;
  background-color: #00A87E;
  border-radius: 4px;
  border-style: solid;
  border-color: #00000000;
  border-width: 1px;
  box-sizing: border-box;
  text-decoration: none;
  justify-content: center;
  display: flex;
  align-items: center;
  line-height: 24px;
  user-select: none;
}

.button:hover {
  background-color: #43B893;
  cursor: pointer;
}

.button:active {
  background-color: #00986A;
}

.text {
  color: #FFFFFF;
  flex: 1;
  font-family: Lato;
  font-size: 14px;
  font-weight: bold;
  text-align: center;
}

.text:hover {
  color: #F7F7F7;
}

.text:active {
  color: #EEEEEE;
}

<div
  className="button"
  onClick={[Function]}
>
  <div
    className="textContainer"
  >
    <span
      className="text"
    >
      test
    </span>
  </div>
</div>
`;

exports[`Button Snapshot Tests Primary small button with icon as LocalSVGSource renderer.creates properly 1`] = `
.textContainer {
  display: inline-flex;
  flex-direction: column;
  vertical-align: middle;
}

.button {
  padding-left: 6px;
  padding-right: 6px;
  height: 32px;
  background-color: #00A87E;
  border-radius: 4px;
  border-style: solid;
  border-color: #00000000;
  border-width: 1px;
  box-sizing: border-box;
  text-decoration: none;
  justify-content: center;
  display: flex;
  align-items: center;
  line-height: 24px;
  user-select: none;
}

.button:hover {
  background-color: #43B893;
  cursor: pointer;
}

.button:active {
  background-color: #00986A;
}

.text {
  color: #FFFFFF;
  flex: 1;
  font-family: Lato;
  font-size: 14px;
  font-weight: bold;
  text-align: center;
}

.text:hover {
  color: #F7F7F7;
}

.text:active {
  color: #EEEEEE;
}

.iconContainer {
  padding-right: 8px;
  display: inline-flex;
  vertical-align: middle;
}

.icon {
  height: 20px;
  width: 20px;
}

<div
  className="button"
  onClick={[Function]}
>
  <div
    className="iconContainer"
  >
    <span
      className="icon"
    >
      <location-fill-pin.svg
        alwaysdark="#121212"
        alwayslight="#FFFFFF"
        basalt="#8A8A8A"
        crystal="#FFFFFF"
        crystalhover="#F7F7F7"
        crystalpressed="#EEEEEE"
        disabled="#FAFAFA"
        glacialblue="#3D98D2"
        granite="#ABABAB"
        height={20}
        larchyellow="#F4BE19"
        larchyellowhover="#F5D624"
        larchyellowpressed="#F3A70C"
        limestone="#E2E2E2"
        limestonehover="#EEEEEE"
        limestonepressed="#D0D0D0"
        obsidian="#121212"
        overlay="rgba(18,18,18,0.3)"
        paintbrushred="#E05666"
        palousegreen="#00A87E"
        palousegreenhighlight="#DEF3EC"
        palousegreenhover="#43B893"
        palousegreenpressed="#00986A"
        quartz="#FAFAFA"
        tintcolor="#FFFFFF"
        width={20}
      />
    </span>
  </div>
  <div
    className="textContainer"
  >
    <span
      className="text"
    >
      has svg
    </span>
  </div>
</div>
`;

exports[`Button Snapshot Tests Secondary large button renderer.creates properly 1`] = `
.textContainer {
  display: inline-flex;
  flex-direction: column;
  vertical-align: middle;
}

.button {
  padding-left: 10px;
  padding-right: 10px;
  height: 40px;
  background-color: #FFFFFF;
  border-radius: 4px;
  border-style: solid;
  border-color: #00A87E;
  border-width: 1px;
  box-sizing: border-box;
  text-decoration: none;
  justify-content: center;
  display: flex;
  align-items: center;
  line-height: 24px;
  user-select: none;
}

.button:hover {
  background-color: #F7F7F7;
  cursor: pointer;
}

.button:active {
  background-color: #EEEEEE;
}

.text {
  color: #00A87E;
  flex: 1;
  font-family: Lato;
  font-size: 16px;
  font-weight: bold;
  text-align: center;
}

.text:hover {
  color: #43B893;
}

.text:active {
  color: #00986A;
}

<div
  className="button"
  onClick={[Function]}
>
  <div
    className="textContainer"
  >
    <span
      className="text"
    >
      testingsecondary
    </span>
  </div>
</div>
`;

exports[`Button Snapshot Tests Secondary small button renderer.creates properly 1`] = `
.textContainer {
  display: inline-flex;
  flex-direction: column;
  vertical-align: middle;
}

.button {
  padding-left: 6px;
  padding-right: 6px;
  height: 32px;
  background-color: #FFFFFF;
  border-radius: 4px;
  border-style: solid;
  border-color: #00A87E;
  border-width: 1px;
  box-sizing: border-box;
  text-decoration: none;
  justify-content: center;
  display: flex;
  align-items: center;
  line-height: 24px;
  user-select: none;
}

.button:hover {
  background-color: #F7F7F7;
  cursor: pointer;
}

.button:active {
  background-color: #EEEEEE;
}

.text {
  color: #00A87E;
  flex: 1;
  font-family: Lato;
  font-size: 14px;
  font-weight: bold;
  text-align: center;
}

.text:hover {
  color: #43B893;
}

.text:active {
  color: #00986A;
}

<div
  className="button"
  onClick={[Function]}
>
  <div
    className="textContainer"
  >
    <span
      className="text"
    >
      testingsecondary
    </span>
  </div>
</div>
`;

exports[`Button Snapshot Tests Tertiary large button renderer.creates properly 1`] = `
.textContainer {
  display: inline-flex;
  flex-direction: column;
  vertical-align: middle;
}

.button {
  padding-left: 10px;
  padding-right: 10px;
  height: 40px;
  background-color: #E2E2E2;
  border-radius: 4px;
  border-style: solid;
  border-color: #00000000;
  border-width: 1px;
  box-sizing: border-box;
  text-decoration: none;
  justify-content: center;
  display: flex;
  align-items: center;
  line-height: 24px;
  user-select: none;
}

.button:hover {
  background-color: #EEEEEE;
  cursor: pointer;
}

.button:active {
  background-color: #D0D0D0;
}

.text {
  color: #121212;
  flex: 1;
  font-family: Lato;
  font-size: 16px;
  font-weight: bold;
  text-align: center;
}

.text:hover {
  color: #121212;
}

.text:active {
  color: #121212;
}

<div
  className="button"
  onClick={[Function]}
>
  <div
    className="textContainer"
  >
    <span
      className="text"
    >
      tert
    </span>
  </div>
</div>
`;

exports[`Button Snapshot Tests Tertiary small button renderer.creates properly 1`] = `
.textContainer {
  display: inline-flex;
  flex-direction: column;
  vertical-align: middle;
}

.button {
  padding-left: 6px;
  padding-right: 6px;
  height: 32px;
  background-color: #E2E2E2;
  border-radius: 4px;
  border-style: solid;
  border-color: #00000000;
  border-width: 1px;
  box-sizing: border-box;
  text-decoration: none;
  justify-content: center;
  display: flex;
  align-items: center;
  line-height: 24px;
  user-select: none;
}

.button:hover {
  background-color: #EEEEEE;
  cursor: pointer;
}

.button:active {
  background-color: #D0D0D0;
}

.text {
  color: #121212;
  flex: 1;
  font-family: Lato;
  font-size: 14px;
  font-weight: bold;
  text-align: center;
}

.text:hover {
  color: #121212;
}

.text:active {
  color: #121212;
}

<div
  className="button"
  onClick={[Function]}
>
  <div
    className="textContainer"
  >
    <span
      className="text"
    >
      tert
    </span>
  </div>
</div>
`;
